vendor/bat-native-ads/src/bat/ads/internal/ad_targeting/ad_targeting.cc

```
@@ -48,4 +48,32 @@ SegmentList AdTargeting::GetSegments() const {
   return segments;
 }
 
+SegmentList AdTargeting::GetInterestSegments() const {
+  SegmentList segments;
+
+  if (features::IsTextClassificationEnabled()) {
+    const ad_targeting::model::TextClassification text_classification_model;
+    const SegmentList text_classification_segments =
+        text_classification_model.GetSegments();
+    segments.insert(segments.end(), text_classification_segments.begin(),
+                    text_classification_segments.end());
+  }
+
+  return segments;
+}
+
+SegmentList AdTargeting::GetIntentSegments() const {
+  SegmentList segments;
+
+  if (features::IsPurchaseIntentEnabled()) {
```

nit: should fail early